networkx/algorithms/coloring/equitable_coloring.py

```diff
@@ -218,15 +218,16 @@ def procedure_P(V_minus, V_plus, N, H, F, C, L, excluded_colors=None):
                         X_prime = U
                         w = v
 
-                        # Finding the solo neighbor of w in X_prime
-                        y_candidates = [
-                            node
-                            for node in L[w]
-                            if F[node] == X_prime and N[(node, W_1)] == 1
-                        ]
-
-                        if len(y_candidates) > 0:
-                            y = y_candidates[0]
+                        try:
+                            # Finding the solo neighbor of w in X_prime
+                            y = next(
+                                node
+                                for node in L[w]
+                                if F[node] == X_prime and N[(node, W_1)] == 1
+                            )
+                        except StopIteration:
+                            pass
+                        else:
```

Comment:
In following block, we have a check `len(y_candidates) > 0` which means that the algorithm allows to list `y_candidates` to be empty. Hence we need to catch `StopIteration` and manage it accordingly. Using the syntax of `try-except-else` syntax [1]:

main branch:
```python
y_candidates = [...]

if len(y_candidates) > 0:
    y = y_candidates[0]
    # list is not empty use first value from list
```

pr:
```python
try:
    y = next(...)
except StopIteration:
    pass
    # iterator is empty skip processing `y`
else:
    # iterator is not empty (exception was not raised) process `y`
```

[1] https://docs.python.org/3/tutorial/errors.html#handling-exceptions